OpenRA.Game/Traits/Util.cs

```diff
@@ -146,5 +146,153 @@ public static int ApplyPercentageModifiers(int number, IEnumerable<int> percenta
 
 			return (int)a;
 		}
+
+		// Algorithm obtained from ftp://ftp.isc.org/pub/usenet/comp.sources.unix/volume26/line3d
+		public static IEnumerable<CPos> Raycast(CPos source, CPos target)
```

Comment:
The most useful form for this would be `public static IEnumerable<CPos> Raycast(WPos source, WPos target)` - return the list of cells that are intesersected by an arbitrary line in world coordinates.